src/reuse.js

```diff
@@ -23,6 +23,19 @@ export const reuse = (unit) => {
     // call
     unitContext.cachedValue = unit();
 
+    // run effects:
+    unitContext.effects.forEach(effect => {
+      if (effect.cleanup) {
+        if (typeof effect.cleanup === 'function') {
+          effect.cleanup();
```

Comment:
What's the reason for doing a cleanup when running the hook again?

I found a bug which originates from this. 
If you unmount the component and mount it back again you will run the cleanup for the effect with no deps (when mounting rather than unmounting)
afterwards it runs the effect again all over.

To reproduce:
1. click toggle
2. Verify that you've seen cleanup for both reuseEffects
3. clear console and click toggle again

Result: no deps reuseEffect cleanup runs and immediately after no deps reuseEffect initializes again.
Expected: cleanup should only run on unmount.

Regardless of this issue, I wonder if reuseEffect should clean up when a component unmounts, do we want to limit effects to run in a component that is always rendered? shouldn't they be more middleware based where they run outside the component lifecycle a la middleware?

Maybe it could be one of the parameters for createStore e.g 
```jsx harmony
const user = () => {
  const [userName, setUsername] = reuseState(null);

  return [username, setUsername];
}

const loginEffect = () => {
  const username = reuse(user);
  reuseEffect(() => {

  }, [username])
}

createStore({ effects: [loginEffect] })
```

Reply:
When thinking about maybe we could allow both ways. something sort of 2 types of effects:

1. persistent effects - always run regardless of the rendering tree
2. regular (standard? transient?) effects  - only runs when the component that used the effect is rendered in the current tree